packages/core/src/components/Input.ts

```diff
@@ -0,0 +1,117 @@
+import { defineComponent, h, ref, computed, watch } from '@vue/runtime-core'
+import chalk from 'chalk'
+import { TuiText } from './Text'
+import { onInputData } from '../composables/input'
+import type { KeyDataEvent } from '../input/types'
+import { useFocus } from '../focus/Focusable'
+
+const SKIP_EVENT_KEY = ['ArrowUp', 'ArrowDown', 'Ctrl', 'Tab', 'Shift']
+
+export const TuiInput = defineComponent({
+  props: {
+    placeholder: {
+      type: String,
+      required: false,
+      default: '',
+    },
+    modelValue: {
+      type: String,
+      required: true,
+    },
+    focus: {
+      type: Boolean,
+      required: true,
+    },
+  },
+  emits: ['update:modelValue'],
+  setup(props, { emit }) {
+    const { active, isFocus } = useFocus({
+      active: props.focus,
+    })
+    const cursorOffset = ref(props.modelValue.length)
+    const content = computed(() => {
+      if (active.value) {
+        if (props.modelValue) {
+          if (
+            props.modelValue &&
+            props.modelValue.length <= cursorOffset.value
+          ) {
+            return props.modelValue + chalk.inverse(' ')
+          }
+
+          const l = props.modelValue.slice(0, cursorOffset.value)
+          const m = chalk.inverse(props.modelValue[cursorOffset.value])
+          const r = props.modelValue.slice(cursorOffset.value + 1)
+
+          return l + m + r
+        } else {
+          return props.placeholder ? '' : chalk.inverse(' ')
+        }
+      } else {
+        return props.modelValue
+      }
+    })
+
+    watch(
+      () => props.focus,
```

Comment:
I'm not so sure about that,Should we try to keep the behavior align with html input tag:

Expose the focus、blur method and let user toggle it with element ref.

Reply:
I think that if we try to align with HTML, it's going to be way too much work for us 😆 
We need to keep the scope a little more reduced, so I think it's fine if we start with less features and then add more.

For this one, I think we can remove the focus prop and only allow keyboard access. Later on we might improve `useFocus()` to add methods like focus and blur